### ISSUE TITLE
Bug Report: Constrained Generation (JSON Schema Grammar) Not Enforced

### ISSUE THREAD

**Issue by sound-of-rain** [Liquid4All/leap-ios#6]
### Summary

When using `setResponseFormat(type:)` with `@Generatable` types, the inference engine recognizes the JSON schema constraint but **does not add the grammar sampler to the sampling chain**, resulting in unconstrained free-text output instead of valid JSON.

### Environment

- LeapSDK versions tested: **0.9.2 and 0.9.3** (both affected)
- Models tested: **LFM2.5-1.2B-Instruct (Q4_0)**, **LFM2-350M (Q4_0)**
- Platforms tested: **iOS device (arm64 with GPU)** and **iOS Simulator (CPU-only, nGpuLayers: 0)**
- Xcode: 26.x
- iOS: 17.0+

### Steps to Reproduce

1. Define a `@Generatable` type:
```swift
@Generatable("Generate creative marketing slogans for a business")
struct SloganResponse: Codable {
  @Guide("A short, catchy slogan") let shortSlogan: String
  @Guide("A longer tagline") let longTagline: String
  @Guide("Key benefits") let keyBenefits: [String]
  @Guide("Target audience") let targetAudience: String
  @Guide("Tone") let tone: String
}
```

2. Use constrained generation:
```swift
var options = GenerationOptions()
try options.setResponseFormat(type: SloganResponse.self)

let stream = conversation.generateResponse(
  message: userMessage, generationOptions: options
)
```

3. Decode response as JSON → **fails** because the output is free text, not JSON.

### Expected Behavior

The model output should be valid JSON matching the schema, enforced by a grammar sampler in the token sampling pipeline.

### Actual Behavior

The model outputs free-text (e.g., `"Absolutely! Here are some engaging slogans..."`) despite the constraint being set.

### Root Cause Evidence

The engine logs show the constraint is **recognized but not enforced**:

```
[ie_llamacpp::text_engine] effective constraint: Some(JsonSchema("{...schema...}"))
common_chat_params_init_lfm2: Using provided grammar
```

However, the **sampler chain does NOT include a grammar step**:
```
[ie_llamacpp::text_engine] sampler info: logits -> penalties -> dry -> top-n-sigma -> top-k -> typical -> top-p -> min-p -> xtc -> temp-ext -> dist
```

If grammar enforcement were working, the chain should include a `grammar` step (e.g., `logits -> grammar -> penalties -> ... -> dist`).

The `common_chat_params_init_lfm2` function logs `"Using provided grammar"` but the grammar sampler is never injected into the actual sampling pipeline.

### Comparison with Fallback (No Constraint)

The fallback generation (without constraints) shows:
```
[ie_llamacpp::text_engine] effective constraint: None
common_chat_params_init_lfm2: Using content relying on the template
```

Same sampler chain — confirming the grammar sampler is absent in both cases.

### Additional Notes

- The `@Generatable` macro and `setResponseFormat()` work correctly — valid JSON schema is generated and passed to the engine
- The binary `InferenceEngineLlamaCppBackend.xcframework` contains all grammar symbols (`llama_grammar_init_impl`, `llama_grammar_accept`, `json_schema_to_grammar`, etc.)
- This affects the SDK's own example apps (LeapSloganExample falls back to regular conversation due to this bug)
- Tested on both physical device (GPU) and simulator (CPU-only) — same behavior
